lxd/networks.go

```diff
@@ -858,6 +863,77 @@ func networkLeasesGet(d *Daemon, r *http.Request) response.Response {
 	return response.SyncResponse(true, leases)
 }
 
+func networkGetLeaseAddresses(s *state.State, network string, hwaddr string) ([]api.InstanceStateNetworkAddress, error) {
+	addresses := []api.InstanceStateNetworkAddress{}
+
+	leaseFile := shared.VarPath("networks", network, "dnsmasq.leases")
+	if !shared.PathExists(leaseFile) {
+		return addresses, nil
+	}
+
+	dbInfo, err := networkLoadByName(s, network)
+	if err != nil {
+		return nil, err
+	}
+
+	content, err := ioutil.ReadFile(leaseFile)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, lease := range strings.Split(string(content), "\n") {
+		fields := strings.Fields(lease)
+		if len(fields) < 5 {
+			continue
+		}
+
+		// Parse the MAC
+		mac := networkGetMacSlice(fields[1])
```

Comment:
@stgraber I'm not quite sure what all of this logic does (or rather why it is needed), however I've found that the `net.ParseMAC` function is quite useful for parsing the leases file in the past (https://github.com/lxc/lxd/blob/master/lxd/device/nic_bridged.go#L933).

It would allow easy case-insensitive comparison if both the argument and the per-line MAC were converted using it, using either its `.String()` function or `bytes.Compare()`.

Reply:
Not sure what it does, I remember dnsmasq having some pretty funky encoding for MACs sometimes, but I don't know if that's what that's supposed to be doing, again, just copy/paste from the other lease function.

Reply:
@stgraber interesting, I tested it this morning with DHCPv6 and it didn't pick it up. Perhaps some DHCPv6 clients generate DUIDs containing the MAC, but apparently not the one in Ubuntu 18.04.